lib/jekyll/site_data/directory.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module SiteData
+    class Directory
+      # Delegate given (zero-arity) method(s) to the Hash object stored in instance
+      # variable `@meta`.
+      def self.delegate_to_meta(*symbols)
+        symbols.each { |sym| define_method(sym) { @meta.send(sym) } }
+      end
+      private_class_method :delegate_to_meta
+
+      delegate_to_meta :freeze, :inspect
+      attr_accessor :context
+
+      def initialize
+        @meta = {}
+      end
+
+      def [](key)
+        @meta[key].tap do |value|
+          value.context = context if value.respond_to?(:context=)
+        end
+      end
+
+      def to_liquid
+        self
```

Comment:
to_liquid is generally a means of sanitizing output, converting into a Drop, or returning a simpler data structure. I don't see a `private` below here so we'd be exposing `#merge`, `#merge!`, and all other Hash methods (from `method_missing`). It seems like we'd rather simply expose the `#[](key)` method to Liquid? I'd rather see a simple Drop here which grants basic access to the underlying Hash of File objects.

Reply:
This was intentional to maintain backwards-compatibility.
As of Jekyll 4.2.x, the top-level objects of *site_data* are `Hash` instances. `Hash#to_liquid` returns the current `Hash` instance. So all of the public methods should continue to be invokable in the future.